maps/reddit.py

```diff
@@ -0,0 +1,111 @@
+import praw
+import click
+import pandas as pd
+import pyarrow as pa
+from nomic import atlas
+from prawcore.exceptions import PrawcoreException
+import time
+
+def get_reddit_instance(client_id, client_secret, user_agent):
+    reddit = praw.Reddit(
+        client_id=client_id,
+        client_secret=client_secret,
+        user_agent=user_agent
+    )
+    return reddit
+
+@click.command()
+@click.option('--client-id', prompt='Enter your Reddit client ID', help='Reddit client ID')
+@click.option('--client-secret', prompt='Enter your Reddit client secret', hide_input=True, help='Reddit client secret')
+@click.option('--user-agent', prompt='Enter your Reddit user agent', help='Reddit user agent')
+@click.option('--reddit-url', prompt='Enter Reddit post URL', help='URL of the Reddit post')
+@click.option('--nomic-api-key', prompt='Enter your Nomic API', help='Nomic API key')
+def main(client_id, client_secret, user_agent, reddit_url, nomic_api_key):
+    reddit = get_reddit_instance(client_id, client_secret, user_agent)
+
+    submission = reddit.submission(url=reddit_url)  # Use submission URL directly
+
+    comments = scrape_reddit_comments(reddit, submission)
+
+    # Convert comments to Apache Arrow table
+    arrow_table = comments_to_arrow_table(comments)
+
+    try:
+        # Generate map name based on Reddit post title
+        map_name = f"Reddit_Post_{submission.title}"
```

Comment:
Is submission.title a snake_case string? If not, maybe we structure the title like 
```[Reddit Post] {submission.title}```